server/routes/pubDocument.tsx

```diff
@@ -207,15 +207,18 @@ const setSurrogateKeysHeadersForPubEdges = async (
 				return acc;
 			}
 
-			const hostname = getCorrectHostname(
-				maybePub.community!.subdomain,
-				maybePub.community!.domain,
-			);
+			const { subdomain, domain } = maybePub.community ?? {};
+
+			const hostname = subdomain ? getCorrectHostname(subdomain, domain) : req.hostname;
+
+			if (acc.includes(hostname)) {
+				return acc;
+			}
 
 			acc.push(hostname);
 			return acc;
 		},
-		[getCorrectHostname(subdomain, domain)] as string[],
+		[req.hostname] as string[],
```

Comment:
i mean this change here: above is how it is on duqduq, below is how it is on prod